### PR TITLE
setting up Mock annotation to be used

### LINKED ISSUE #166: Make @Mock annotation available
Students should be able to use `@Mock` annotation from Mockito.

### PR DESCRIPTION
**Pull request by AtillaColak**
I updated the tests accordingly and they passed. They were some irrelevant failures on my local about weblab credentials not logged in, so I disregarded it since it's not related to this error. Please let me know if more changes have to be done to keep consistency. This MR closes #166

### CODE REVIEW

**Review by mauricioaniche — APPROVED** [review]
This looks great to me! Super neat and clean code. Congrats!

I wouldn't expect tests to fail in local. Are they marked as CI tests? They shouldn't run locally. I'll put the pipeline to run here, let's see if everything passes!

File: andy/src/main/java/nl/tudelft/cse1110/andy/codechecker/checks/MockClass.java [hunk 1]
```diff
@@ -38,6 +41,21 @@ public boolean visit(MethodInvocation mi) {
...
+    @Override
+    public boolean visit(FieldDeclaration fd) {
+        boolean hasMockAnnotation = fd.modifiers().stream()
+                .anyMatch(m -> m instanceof Annotation &&
+                        ((Annotation)m).getTypeName().getFullyQualifiedName().equals("Mock"));
+
+        // If the field is annotated with @Mock, check if it's the class we are interested in
+        if (hasMockAnnotation) {
+            String className = fd.getType().toString();
+            classWasMocked = className.contains(classToBeMocked);
```
Comment (martinmladenov) [inline andy/src/main/java/nl/tudelft/cse1110/andy/codechecker/checks/MockClass.java:53]: If I understand the code correctly, `classWasMocked` could be erroneously set to false if there are multiple usages of `@Mock`.  For example, the following could happen:
1. A search for a mocked instance of `Set` is started
2. `@Mock Set<Integer> a;` is visited,  `classWasMocked` is set to true
3. `@Mock List<String> b;` is visited,  `classWasMocked` is set to false
4. The final returned value is false

I see that the other visitor prevents this problem by only executing the logic if `classWasMocked` is still false.

**Review by martinmladenov — CHANGES_REQUESTED** [review]

File: andy/src/main/java/nl/tudelft/cse1110/andy/codechecker/checks/MockClass.java (hunk 1 above)
Comment (AtillaColak) [reply andy/src/main/java/nl/tudelft/cse1110/andy/codechecker/checks/MockClass.java:53]: Yeah, you're right. To handle that, I should add a `(!classWasMocked)` conditional. I will update the code and tests respectively.

Comment (AtillaColak) [reply andy/src/main/java/nl/tudelft/cse1110/andy/codechecker/checks/MockClass.java]: Updates committed and new test passes.

**Review by martinmladenov — APPROVED** [review]

File: andy/src/main/java/nl/tudelft/cse1110/andy/codechecker/checks/MockClass.java (hunk 1 above)
Comment (martinmladenov) [reply andy/src/main/java/nl/tudelft/cse1110/andy/codechecker/checks/MockClass.java]: Thanks! :)

**Pull request merged**